src/api/nlopt-in.hpp

```diff
@@ -299,6 +343,14 @@ namespace nlopt {
       mythrow(nlopt_set_min_objective(o, myvfunc, d)); // d freed via o
       alloc_tmp();
     }
+    void set_min_objective(functor_type functor) {
+      myfunc_data *d =
+        reinterpret_cast<myfunc_data*>(alloc_myfunc_data_with_nulls());
+
+      d->functor = std::move(functor);
+      mythrow(nlopt_set_min_objective(o, functor_wrapper, d)); // d freed via o
+    }
+
     void set_max_objective(func f, void *f_data) {
       myfunc_data *d = new myfunc_data;
```

Comment:
Shouldn't this (and similar cases) now be `myfunc_data *d = new myfunc_data()` to zero-initialize now, to be sure that the `functor` field gets a default initializer?

Reply:
Not if you zero-initialize manually the rest of the fields (which are raw pointers): since `std::function` is an STL container, it has a default constructor, which is called when `myfunc_data` is default-initialized (or zero-initialized), so `functor` will never be in an undefined state.

But what I could do instead, is rewrite those methods with `alloc_myfunc_data_with_nulls` and not deal with `new` explicitly in any of them.

Reply:
okay, done! (: